presto-main/src/main/java/com/facebook/presto/operator/repartition/OptimizedPartitionedOutputOperator.java

```diff
@@ -224,6 +226,32 @@ static DecodedBlockNode decodeBlock(BlockFlattener flattener, Closer blockLeaseC
         return new DecodedBlockNode(decodedBlock, ImmutableList.of());
     }
 
+    static int[] ensureCapacityWithAllocator(int[] buffer, int capacity, ExpansionFactor expansionFactor, Arrays.ExpansionOption expansionOption, ArrayAllocator allocator)
+    {
+        int newCapacity = (int) (capacity * expansionFactor.getExpansionFactor());
+
+        int[] newBuffer;
+        if (buffer == null) {
+            newBuffer = allocator.borrowIntArray(newCapacity);
+        }
+        else if (buffer.length < capacity) {
+            newBuffer = allocator.borrowIntArray(newCapacity);
+            if (expansionOption == PRESERVE) {
+                System.arraycopy(buffer, 0, newBuffer, 0, Math.min(buffer.length, capacity));
```

Comment:
Math.min(buffer.length, capacity) -> buffer.length since buffer.length is known to be less than capacity